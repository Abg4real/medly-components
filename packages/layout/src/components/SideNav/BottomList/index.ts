import { BottomList } from './BottomList.styled';

export default BottomList;
