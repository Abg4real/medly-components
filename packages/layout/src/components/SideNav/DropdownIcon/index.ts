import { DropdownIcon } from './DropdownIcon.styled';

export default DropdownIcon;
