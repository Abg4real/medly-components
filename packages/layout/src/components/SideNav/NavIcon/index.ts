import { NavIcon } from './NavIcon.styled';

export default NavIcon;
