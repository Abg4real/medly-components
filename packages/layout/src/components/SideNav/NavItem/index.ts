import { NavItemStyled as NavItem } from './NavItem.styled';

export default NavItem;
