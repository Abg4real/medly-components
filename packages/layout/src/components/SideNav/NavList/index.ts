import { NavListStyled as NavList } from './NavList.styled';

export default NavList;
