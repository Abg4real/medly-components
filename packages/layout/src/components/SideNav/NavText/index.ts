import { NavText } from './NavText';

export default NavText;
