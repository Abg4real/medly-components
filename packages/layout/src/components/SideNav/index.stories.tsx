import { Text } from '@medly-components/core';
import { EditIcon, HomeIcon, SearchIcon } from '@medly-components/icons';
import { storiesOf } from '@storybook/react';
import React from 'react';
import SideNav from './SideNav';

storiesOf('Layout', module).add(
    'SideNav',
    () => (
        <SideNav>
            <SideNav.NavItem as={'a'} href="#">
                <SideNav.NavIcon>
                    <HomeIcon />
                </SideNav.NavIcon>
                <SideNav.NavText>Home</SideNav.NavText>
            </SideNav.NavItem>
            <SideNav.NavItem openSideNavOnClick active>
                <SideNav.NavIcon>
                    <SearchIcon />
                </SideNav.NavIcon>
                <SideNav.NavText>Search</SideNav.NavText>
                <SideNav.SubNavList>
                    <SideNav.NavItem active>
                        <SideNav.NavText>Cars</SideNav.NavText>
                    </SideNav.NavItem>
                    <SideNav.NavItem>
                        <SideNav.NavText>Bikes</SideNav.NavText>
                    </SideNav.NavItem>
                    <SideNav.NavItem>
                        <SideNav.NavText>Phones</SideNav.NavText>
                    </SideNav.NavItem>
                </SideNav.SubNavList>
            </SideNav.NavItem>
            <SideNav.NavItem>
                <SideNav.NavIcon>
                    <EditIcon />
                </SideNav.NavIcon>
                <SideNav.NavText>Edit</SideNav.NavText>
            </SideNav.NavItem>
        </SideNav>
    ),
    {
        props: {
            propTablesExclude: [Text, SideNav.NavItem]
        }
    }
);
