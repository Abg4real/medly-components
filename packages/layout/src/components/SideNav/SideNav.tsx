import { Text } from '@medly-components/core';
import { BurgerIcon } from '@medly-components/icons';
import { useOuterClickNotifier, WithStyle } from '@medly-components/utils';
import React, { SFC, useRef, useState } from 'react';
import NavItem from './NavItem';
import NavList from './NavList';
import * as Styled from './SideNav.styled';
import SubNavList from './SubNavList';
import { SideNavProps, SideNavStaticProps } from './types';

export const SideNav: SFC<SideNavProps> & WithStyle & SideNavStaticProps = props => {
    const ref = useRef(null);
    const [open, setOpenState] = useState(false),
        Logo = props.logo;

    const burgerIconClickHandler = () => {
            setOpenState(!open);
        },
        sidenavOpenHandler = () => {
            !open && setOpenState(true);
        };

    useOuterClickNotifier(() => {
        props.closeOnOuterClick && setOpenState(false);
    }, ref);

    return (
        <Styled.SideNav open={open} ref={ref} position="left" data-testid="sidenav">
            <NavItem as="div">
                <Styled.Icon>
                    <BurgerIcon size="S" open={open} onClick={burgerIconClickHandler} />
                </Styled.Icon>
                <Styled.Logo>{Logo && <Logo />}</Styled.Logo>
            </NavItem>
            {React.Children.map(props.children, child => {
                return React.cloneElement(child as any, { sidenavOpenHandler });
            })}
        </Styled.SideNav>
    );
};

SideNav.BottomList = Styled.BottomList;
SideNav.NavList = NavList;
SideNav.SubNavList = SubNavList;
SideNav.NavItem = NavItem;
SideNav.NavIcon = Styled.Icon;
SideNav.NavText = Text;
SideNav.Logo = Styled.Logo;

SideNav.displayName = 'SideNav';
SideNav.Style = Styled.SideNav;
SideNav.defaultProps = {
    closeOnOuterClick: false
};
