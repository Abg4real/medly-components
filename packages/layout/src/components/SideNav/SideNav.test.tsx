import { AddIcon, HomeIcon, SearchIcon } from '@medly-components/icons';
import { TestUtils } from '@medly-components/utils';
import React from 'react';
import SideNav from './SideNav';

jest.mock('react-datepicker/dist/react-datepicker.css', () => ({}));

const { act, render, fireEvent, cleanup } = TestUtils;
afterEach(cleanup);

describe('SideNav', () => {
    it('should render properly', () => {
        const { container } = render(
            <SideNav logo={AddIcon}>
                <SideNav.NavList>
                    <SideNav.NavItem active>
                        <SideNav.NavIcon>
                            <SearchIcon />
                        </SideNav.NavIcon>
                        <SideNav.NavText>Search</SideNav.NavText>
                    </SideNav.NavItem>
                </SideNav.NavList>
                <SideNav.BottomList>
                    <SideNav.NavItem>
                        <SideNav.NavIcon>
                            <HomeIcon />
                        </SideNav.NavIcon>
                        <SideNav.NavText>Home</SideNav.NavText>
                    </SideNav.NavItem>
                </SideNav.BottomList>
            </SideNav>
        );
        expect(container).toMatchSnapshot();
    });

    it('should change width after clicking on burger icon', () => {
        const { container, getByTestId } = render(
            <SideNav>
                <SideNav.NavList>
                    <SideNav.NavItem>
                        <SideNav.NavIcon>
                            <HomeIcon />
                        </SideNav.NavIcon>
                        <SideNav.NavText>Home</SideNav.NavText>
                    </SideNav.NavItem>
                    <SideNav.NavItem active>
                        <SideNav.NavIcon>
                            <SearchIcon />
                        </SideNav.NavIcon>
                        <SideNav.NavText>Search</SideNav.NavText>
                    </SideNav.NavItem>
                </SideNav.NavList>
            </SideNav>
        );
        act(() => {
            fireEvent.click(container.querySelector('button'));
        });
        expect(getByTestId('sidenav')).toMatchSnapshot();
    });

    it('should change width when clicked on item with sidenavOpenCloseOnClick prop given', () => {
        const mockOnClick = jest.fn();

        const { getByText, getByTestId } = render(
            <SideNav>
                <SideNav.NavList>
                    <SideNav.NavItem>
                        <SideNav.NavIcon>
                            <HomeIcon />
                        </SideNav.NavIcon>
                        <SideNav.NavText>Home</SideNav.NavText>
                    </SideNav.NavItem>
                    <SideNav.NavItem active openSideNavOnClick onClick={mockOnClick}>
                        <SideNav.NavIcon>
                            <SearchIcon />
                        </SideNav.NavIcon>
                        <SideNav.NavText>Search</SideNav.NavText>
                        <SideNav.SubNavList>
                            <SideNav.NavItem active>
                                <SideNav.NavText>Cars</SideNav.NavText>
                            </SideNav.NavItem>
                            <SideNav.NavItem>
                                <SideNav.NavText>Bikes</SideNav.NavText>
                            </SideNav.NavItem>
                            <SideNav.NavItem>
                                <SideNav.NavText>Phones</SideNav.NavText>
                            </SideNav.NavItem>
                        </SideNav.SubNavList>
                    </SideNav.NavItem>
                </SideNav.NavList>
            </SideNav>
        );
        act(() => {
            fireEvent.click(getByText('Search'));
        });
        expect(mockOnClick).toBeCalled();
        expect(getByTestId('sidenav')).toMatchSnapshot();
    });

    it('should close sidenav if we click outside', () => {
        const { container, getByText, getByTestId } = render(
            <div>
                <p>Outer Element</p>
                <SideNav>
                    <SideNav.NavList>
                        <SideNav.NavItem>
                            <SideNav.NavIcon>
                                <HomeIcon />
                            </SideNav.NavIcon>
                            <SideNav.NavText>Home</SideNav.NavText>
                        </SideNav.NavItem>
                    </SideNav.NavList>
                </SideNav>
            </div>
        );
        act(() => {
            fireEvent.click(container.querySelector('button'));
        });
        act(() => {
            fireEvent.click(getByText('Outer Element'));
        });
        expect(getByTestId('sidenav')).toMatchSnapshot();
    });
});
