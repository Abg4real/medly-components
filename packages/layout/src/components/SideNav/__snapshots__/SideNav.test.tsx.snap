// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render properly 1`] = `
.c5 {
  overflow: visible;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  font-size: 2.4rem;
  cursor: inherit;
}

.c5 * {
  fill-opacity: 1;
  fill: #607890;
}

.c7 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c3 {
  width: 100%;
  border-left: 0.4rem solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: calc(7.2rem - 0.8rem) calc(25.6rem - 7.2rem - 1.6rem);
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  cursor: pointer;
  border-color: #ffffff;
}

.c3 .c4 {
  overflow: visible;
  font-size: 2.4rem;
}

.c3 .c6 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  opacity: 0;
  font-weight: 600;
  color: #005AEE;
}

.c3 .c4 {
  padding: calc((4.8rem - 2.4rem) / 2);
  border-radius: 50%;
  background-color: rgba(0,90,238,0.1);
}

.c3 .c4 * {
  fill: #005AEE;
}

.c8 {
  width: 100%;
  border-left: 0.4rem solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: calc(7.2rem - 0.8rem) calc(25.6rem - 7.2rem - 1.6rem);
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  cursor: pointer;
  border-color: #ffffff;
}

.c8 .c4 {
  overflow: visible;
  font-size: 2.4rem;
}

.c8 .c6 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  opacity: 0;
  font-weight: 600;
  color: #607890;
}

.c8 .c4 {
  padding: calc((4.8rem - 2.4rem) / 2);
  border-radius: 50%;
}

.c8 .c4:hover {
  background-color: #eff2f4;
}

.c8 .c4:hover * {
  fill: #435465;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 2rem 0 0 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
}

.c2::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c2::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c0 {
  height: 100vh;
  background-color: #dfe4e9;
  width: 0;
  position: fixed;
  z-index: 10;
  top: 0;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  left: 0;
  position: inherit;
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  width: 7.2rem;
  box-shadow: 0.2rem 0 8px rgba(96,120,144,0.2);
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 200ms ease-out,box-shadow 200ms ease-out;
  transition: width 200ms ease-out,box-shadow 200ms ease-out;
  background-color: #ffffff;
  padding-right: 0;
  width: 7.2rem;
  box-shadow: none;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 2rem 0 0 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
  min-height: 7.2rem;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  position: relative;
  width: 100%;
}

.c9::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c9::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c9::before {
  content: '';
  width: calc(100% - 3.2rem);
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 1px solid #dfe4e9;
  top: 0;
  left: 0;
  right: 0;
}

<div>
  <aside
    class="sc-pJkiN c0"
    id="medly-sidenav"
  >
    <nav
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3"
          font-variant="body1"
        >
          <svg
            class="c4 c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.998 19.328v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87l-8.36-7.53c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87h1.7v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <span
            class="c6 c7"
          >
            Home
          </span>
        </li>
        <li
          class="c8"
          font-variant="body1"
        >
          <svg
            class="c4 c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M15.187 14.472h.79l4.24 4.26c.41.41.41 1.08 0 1.49-.41.41-1.08.41-1.49 0l-4.25-4.25v-.79l-.27-.28a6.5 6.5 0 0 1-5.34 1.48c-2.78-.47-5-2.79-5.34-5.59a6.505 6.505 0 0 1 7.27-7.27c2.8.34 5.12 2.56 5.59 5.34a6.5 6.5 0 0 1-1.48 5.34l.28.27zm-9.71-4.5c0 2.49 2.01 4.5 4.5 4.5s4.5-2.01 4.5-4.5-2.01-4.5-4.5-4.5-4.5 2.01-4.5 4.5z"
              fill="#000"
              fill-opacity=".54"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="c6 c7"
          >
            Search
          </span>
        </li>
      </ul>
      <ul
        class="c9"
      >
        <li
          class="c8"
          font-variant="body1"
        >
          <svg
            class="c4 c5"
            fill="none"
            height="1em"
            id="medly-sidenav-toggle-expand"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.111 6H1.89C1.4 6 1 6.45 1 7s.4 1 .889 1H16.11C16.6 8 17 7.55 17 7s-.4-1-.889-1zM16.111 11H1.89C1.4 11 1 11.45 1 12s.4 1 .889 1H16.11c.489 0 .889-.45.889-1s-.4-1-.889-1zM17 17c0-.55-.4-1-.889-1H1.89C1.4 16 1 16.45 1 17s.4 1 .889 1H16.11c.489 0 .889-.45.889-1zM20.176 15.211l2.537-2.525a.967.967 0 0 0 0-1.374l-2.537-2.525c-.618-.615-1.676-.176-1.676.692v5.05c0 .867 1.058 1.296 1.676.682z"
              fill="#000"
            />
          </svg>
          <span
            class="c6 c7"
          >
            Open Menu
          </span>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
